### PR TITLE
RESKC-208:Use custom mapper to fix peristence mappings

### PR DESCRIPTION
**Pull request by blackcathacker**
Instead of trying to use default mappings and custom converters, use custom mapper so xstream will seriazlie and deserialize all proxied/managed lists as real lists.

There is still duplicate code in XmlObjectSerializationServiceImpl that I'd like to remove, but this seems sufficient to resolve all known issues. Works with PD, Question and PeopleFlow which encompasses all known problem areas.

### CODE REVIEW

**Comment by leoherbie** [comment]
There are other Jpa indirect collection types.  While we are changing this thing we should support those too.  I believe rice is using a Set in at least one place which would map to an indirect set

**Comment by blackcathacker** [comment]
So just IndirectMap and IndirectSet as well?

**Comment by leoherbie** [comment]
Think so.  Not at my computer

**Comment by blackcathacker** [comment]
Updated to include other JPA indirect classes.

File: rice-framework/krad-service-impl/src/main/java/org/kuali/rice/krad/service/impl/SerializerServiceBase.java [hunk 1]
```diff
@@ -207,7 +185,70 @@ public boolean canConvert(Class clazz) {
...
+    			Class manageableArray = Class.forName("org.apache.ojb.broker.util.collections.ManageableArrayList");
+    			collectionMap.put(manageableArray, ArrayList.class);
+    		} catch (Exception ex) {
+    			//do nothing if classes cannot be registered
+    		}
+    		try {
+    			Class jpaProxy = Class.forName("org.eclipse.persistence.indirection.IndirectList");
+    			collectionMap.put(jpaProxy, ArrayList.class);
+    			jpaProxy = Class.forName("org.eclipse.persistence.indirection.IndirectMap");
+    			collectionMap.put(jpaProxy, Hashtable.class);
```
Comment (leoherbie) [inline rice-framework/krad-service-impl/src/main/java/org/kuali/rice/krad/service/impl/SerializerServiceBase.java]: You want a HashMap here

File: rice-framework/krad-service-impl/src/main/java/org/kuali/rice/krad/service/impl/SerializerServiceBase.java [hunk 2]
```diff
@@ -207,7 +185,70 @@ public boolean canConvert(Class clazz) {
...
+    			jpaProxy = Class.forName("org.eclipse.persistence.indirection.IndirectSet");
+    			collectionMap.put(jpaProxy, Set.class);
```
Comment (leoherbie) [inline rice-framework/krad-service-impl/src/main/java/org/kuali/rice/krad/service/impl/SerializerServiceBase.java]: you want a HashSet here

File: rice-framework/krad-service-impl/src/main/java/org/kuali/rice/krad/service/impl/SerializerServiceBase.java (hunk 1 above)
Comment (blackcathacker) [inline rice-framework/krad-service-impl/src/main/java/org/kuali/rice/krad/service/impl/SerializerServiceBase.java]: So I based these on which classes the indirect versions implement or extend. Happy to change if you think though. Thoughts?

Comment (leoherbie) [inline rice-framework/krad-service-impl/src/main/java/org/kuali/rice/krad/service/impl/SerializerServiceBase.java]: I think we would want to choose a concrete implementation and so I would probably choose HashMap for IndirectMap.  As for the IndirectSet I would think you will need to use a non-abstract class so HashSet would be the ideal choice there.  This is just a guess as I don't fully know what xstream does if you pass it an interface or abstract class type.

**Comment by blackcathacker** [comment]
Updated to use HashMap and HashSet.

**Pull request merged**